packages/core/src/middleware/koa-serve-custom-ui-assets.ts

```diff
@@ -29,12 +35,48 @@ export default function koaServeCustomUiAssets(customUiAssetId: string) {
     const isExisted = await isFileExisted(fileObjectKey);
     assertThat(isExisted, 'entity.not_found', 404);
 
-    const downloadResponse = await downloadFile(fileObjectKey);
-    ctx.type = downloadResponse.contentType ?? 'application/octet-stream';
-    ctx.body = downloadResponse.readableStreamBody;
+    const range = ctx.get('range');
+    const { start, end, count } = parseRange(range);
+
+    const [
+      { contentLength = 0, readableStreamBody, contentType },
+      { contentLength: totalFileSize = 0 },
+    ] = await Promise.all([
+      downloadFile(fileObjectKey, start, count),
+      getFileProperties(fileObjectKey),
+    ]);
+
+    ctx.body = readableStreamBody;
+    ctx.type = contentType ?? 'application/octet-stream';
+    ctx.status = range ? 206 : 200;
+
+    ctx.set('Cache-Control', isFileRequest ? maxAgeSevenDays : noCache);
```

Comment:
nit: the definition of `isFileRequest` is a little bit vague since index.html is also a file.

Reply:
plus, we should check if it's file request by checking the last portion of the path? i.e. `path.basename()`

Reply:
`path.basename()` sometimes does not work as expected. Do you remember last time we were dealing with a google connector callback URL with query params that contains dot in the tunnel CLI? We tried `path.basename()` and it wasnt't able to handle that case.

So I updated the code and copied the `isFileAssetPath` function from tunnel CLI to core-kit. Now both the tunnel CLI and the core middleware will use the core-kit util. Will release a new version of core-kit and tunnel CLI separately, too.